tests/data_context/migrator/test_cloud_migrator.py

```diff
@@ -161,6 +109,91 @@ def _build_mock_migrate(
     return _build_mock_migrate
 
 
+@pytest.fixture
+def assert_stdout_is_accurate_and_properly_ordered() -> Callable:
+    def _assert(stdout: str, statements: List[str]) -> None:
+        last_position = -1
+        for statement in statements:
+            position = stdout.find(statement)
+            assert position != -1, f"Could not find '{statement}' in stdout"
+            assert (
+                position > last_position
+            ), f"Statement '{statement}' occurred in the wrong order"
+
+    return _assert
```

Comment:
When do we want global funcs and when do we want fixtures? I tend to lean towards the latter here because I want the test to show an explicit dependency on this helper.

Reply:
Good question. I don't know if there is a hard and fast rule, but I usually use fixtures that generate a function when I am basically using that function to generate what would normally be passed in as a fixture based on params passed into the generated fn during the test. In other words, if the fixture fn returns something I would have passed in as a fixture, I create the fn as a fixture.

I use a global function when I am using it to encapsulate some repeated assertions like you are doing here.

I think the difference in my mind is that a fixture is more a part of the setup of the test and the global fns are part of the verify of the "setup" "evaluate" "verify"

Reply:
going to change to make this a global func